pkg/remote/codec/thrift/thrift.go

```diff
@@ -82,7 +82,8 @@ func (c thriftCodec) Marshal(ctx context.Context, message remote.Message, out re
 
 	// 2. encode thrift
 	// encode with hyper codec
-	if c.hasHyperMarshal(data) {
+	// NOTE: ensure hyperMarshalEnabled is inlined, or it may cause performance loss
```

Comment:
NOTE: to ensure hyperMarshalEnabled is inlined so split the check logic, or it may cause performance loss